icu4c/source/i18n/translit.cpp

```diff
@@ -1508,28 +1508,29 @@ UBool Transliterator::initializeRegistry(UErrorCode &status) {
      */
     //static const char translit_index[] = "translit_index";
 
+    UErrorCode lstatus = U_ZERO_ERROR;
     UResourceBundle *bundle, *transIDs, *colBund;
-    bundle = ures_open(U_ICUDATA_TRANSLIT, NULL/*open default locale*/, &status);
-    transIDs = ures_getByKey(bundle, RB_RULE_BASED_IDS, 0, &status);
+    bundle = ures_open(U_ICUDATA_TRANSLIT, NULL/*open default locale*/, &lstatus);
+    transIDs = ures_getByKey(bundle, RB_RULE_BASED_IDS, 0, &lstatus);
     const UnicodeString T_PART = UNICODE_STRING_SIMPLE("-t-");
 
     int32_t row, maxRows;
-    if (U_SUCCESS(status)) {
```

Comment:
>if (U_SUCCESS(status)) { [](start = 4, length = 24)

The code before doesn't really do a great job of checking if the data loading above fails due to out-of-memory (OOM) -- but with the changes to use a local status variable now (`lstatus`) any OOM failure from `ures_open` or `ures_getByKey` would be completely hidden entirely.

Would you mind adding a check to see if `lstatus` is OOM here? 

For example, here's a similar example in another file where a localStatus is used:
[https://github.com/unicode-org/icu/blob/master/icu4c/source/i18n/numsys.cpp#L153](https://github.com/unicode-org/icu/blob/master/icu4c/source/i18n/numsys.cpp#L153)

Perhaps something like this:
```
if (lstatus == U_MEMORY_ALLOCATION_ERROR) {
    delete registry;
    registry = nullptr;
    status = U_MEMORY_ALLOCATION_ERROR;
    return FALSE;
}
```